card/mod/01_core/set/all/pattern.rb

```diff
@@ -19,26 +19,28 @@ def reset_patterns_if_rule saving=false
     set.reset_patterns
     set.include_set_modules
 
-    #this is really messy.
-    if saving
-      self.add_to_read_rule_update_queue( set.item_cards limit: 0 ) if right.id == Card::ReadID
+    # FIXME: should be in right/read.rb
+    if saving && right.id == Card::ReadID
+      self.add_to_read_rule_update_queue(set.item_cards limit: 0)
     end
   end
 end
 
 def safe_set_keys
-  patterns.map( &:safe_key ).reverse * " "
+  patterns.map(&:safe_key).reverse * " "
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.